src/main/java/org/sagebionetworks/web/client/view/ProfileViewImpl.java

```diff
@@ -510,13 +507,20 @@ public void onClick(ClickEvent event) {
 	}
 	
 	@Override
-	public void setTeams(List<Team> teams, boolean isOwner) {
-		myTeamsWidget.configure(teams, false, isOwner, new TeamListWidget.RequestCountCallback() {
-			@Override
-			public void invoke(String teamId, Long requestCount) {
-				presenter.addMembershipRequests(requestCount.intValue());
-			}
-		});
+	public void setTeams(List<Team> teams) {
+		myTeamsWidget.configure(teams, false);
+	}
+	
+	@Override
+	public void setTeamsFromBundle(List<TeamRequestBundle> teamsRequestBundles) {
+		int requestCount = 0;
+		List<Team> teams = new ArrayList<Team>(teamsRequestBundles.size());
+		for (TeamRequestBundle teamAndRequest: teamsRequestBundles) {
+			requestCount += teamAndRequest.getRequestCount();
```

Comment:
request count should be calculated in the presenter (and add to the total team notification count) instead of adding it up in the view and communicating it back to the presenter.